Shopify/cli
# Fetch partners account information

### WHY are these changes introduced?

partially fixes: https://github.com/Shopify/develop-app-management/issues/17
partially fixes: https://github.com/Shopify/develop-app-management/issues/1474

NOTE: The `partners` [PR](https://github.com/Shopify/partners/pull/51366) should be shipped first



### WHAT is this pull request doing?
- Fetch partners current account information when the partners access token is granted, either being a user account or a service account. Now account information is available to be used at the very first stage of the command flow
- Refactor every method that ends calling a `findOrg` operation. Instead of receiving the `token` it's going to receive the new `partners session` object that includes the `token` inside plus other relevant session info
- Display the account information in the `app info` command. 
- Display the account information inside the no org found error banner. Besides it's also added a suggestion to run `auth logout` command 


### How to test your changes?
- Create a spin instance that extends partners and includes business-platform
```sh
spin up partners:agencies
```
_User account_
- Run the command `app info` and the user account should be visible
```sh
NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) p shopify app info --path  /your/app/path
```
<img src="https://github.com/Shopify/cli/assets/105213827/f93e09dd-c19e-4ebe-bc11-fa0d5a855cfb" width=400/>



- Run the command `dev command` using an account that has now organization created yet
```sh
NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) p shopify app dev --path  /your/app/path
```


<img src="https://github.com/Shopify/cli/assets/105213827/dac5f0d4-620b-48f3-96a1-43ef1baeaf99" width=600/>

_Partners account_
- First you need to access the partners dashboard and create the token
- Run the command `app info` and the service account should be visible
```sh
SHOPIFY_CLI_PARTNERS_TOKEN=<YOUR_TOKEN> NODE_TLS_REJECT_UNAUTHORIZED=0 SHOPIFY_SERVICE_ENV=spin SPIN_INSTANCE=$(spin show -o name) p shopify app info --path  /your/app/path
```
![image](https://github.com/Shopify/cli/assets/105213827/90290597-e0da-4f11-a92c-df8757bb65e0)




### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
